tests/managed_services/mas_debezium/conftest.py

```diff
@@ -0,0 +1,250 @@
+        async_req=True, _async=_async, id=kafka_create_api.id
+    )
+    kafka_list_samples = TimeoutSampler(
+        wait_timeout=WAIT_STATUS_TIMEOUT,
+        sleep=10,
+        func=lambda: kafka_mgmt_api_instance.get_kafkas(search=f"name = {KAFKA_NAME}"),
+    )
+    kafka_list_sample = None
+    try:
+        for kafka_list_sample in kafka_list_samples:
+            if kafka_list_sample.size == 0:
+                break
+    except TimeoutExpiredError:
+        LOGGER.error(
+            f"Timeout expired for deleting {KAFKA_NAME} kafka instance:\n"
+            f"{kafka_list_sample}"
+        )
+        raise
+
+
+@pytest.fixture(scope="class")
+def kafka_instance_ready(kafka_mgmt_api_instance, kafka_instance):
+    LOGGER.info(f"Waiting for {KAFKA_NAME} kafka instance to be ready")
+    kafka_samples = TimeoutSampler(
+        wait_timeout=400,
+        sleep=10,
+        func=lambda: kafka_mgmt_api_instance.get_kafka_by_id(id=kafka_instance.id),
+    )
+    kafka_sample = None
+    try:
+        for kafka_sample in kafka_samples:
+            if kafka_sample.status == "ready":
+                return kafka_sample
+    except TimeoutExpiredError:
+        LOGGER.error("Timeout expired. Current kafka snapshot:\n" f"{kafka_sample}")
+        raise
+
+
+@pytest.fixture(scope="class")
+def kafka_instance_client(kafka_instance_ready, access_token):
+    # https://github.com/redhat-developer/app-services-sdk-python/tree/main/sdks/kafka_instance_sdk
+    configuration = rhoas_kafka_instance_sdk.Configuration(
+        host=kafka_instance_ready.admin_api_server_url, access_token=access_token
+    )
+
+    with rhoas_kafka_instance_sdk.ApiClient(
+        configuration=configuration
+    ) as kafka_api_client:
+        yield kafka_api_client
+
+
+@pytest.fixture(scope="class")
+def kafka_instance_sa(kafka_instance_client, service_accounts_api_instance):
+    LOGGER.info(
+        f"Creating {KAFKA_SA_NAME} service-account for {KAFKA_NAME} kafka instance"
+    )
+    service_account_create_request_data = ServiceAccountCreateRequestData(
+        name=KAFKA_SA_NAME,
+        description=f"{KAFKA_NAME} instance service-account",
+    )
+    kafka_sa = service_accounts_api_instance.create_service_account(
+        service_account_create_request_data=service_account_create_request_data
+    )
+    assert (
+        kafka_sa.id
+    ), f"Failed to create service-account for kafka. API response:\n{kafka_sa}"
+
+    yield kafka_sa
+
+    LOGGER.info(f"Waiting for {KAFKA_SA_NAME} service-account to be deleted")
+    service_accounts_api_instance.delete_service_account(id=kafka_sa.id, async_req=True)
+    sa_list_samples = TimeoutSampler(
+        wait_timeout=WAIT_STATUS_TIMEOUT,
+        sleep=10,
+        func=lambda: service_accounts_api_instance.get_service_accounts(
+            client_id=[kafka_sa.id]
+        ),
+    )
+    sa_list_sample = None
+    try:
+        for sa_list_sample in sa_list_samples:
+            if len(sa_list_sample) == 0:
```

Comment:
`if len(sa_list_sample) == 0:` if sa_list_sample is `None` you will get error
`TypeError: object of type 'NoneType' has no len()`

Reply:
changed `sa_list_sample` to `N/A`; done